src/mmw/js/src/compare/views.js

```diff
@@ -41,6 +42,18 @@ var CompareWindow = Marionette.LayoutView.extend({
         // so the offset of the container needs to be
         // recomputed.
         $(window).bind('resize.app', _.debounce(_.bind(this.updateContainerPos, this)));
+        this.listenTo(App.user, 'change:guest', this.saveAfterLogin);
+    },
+
+    saveAfterLogin: function(user, guest) {
+        if (!guest && this.model.isNew()) {
+            var user_id = user.get('id');
+            this.model.set('user_id', user_id);
+            this.model.get('scenarios').each(function(scenario) {
+                scenario.set('user_id', user_id);
```

Comment:
This seems to be a replica of [`saveAfterLogin`](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/modeling/views.js#L69-L78) in `modeling/views.js`. That line, originally added in #385, was placed outside of `saveProjectAndScenarios` since we only need to set it for saving existing projects and scenarios that don't have a user assigned to them. Every project and scenario created when a user is logged in already has this set, so doing so every time in `saveProjectAndScenarios` would be redundant.